src/sagemaker/jumpstart/types.py

```diff
@@ -940,10 +1004,340 @@ def to_json(self) -> Dict[str, Any]:
+
+
+class JumpStartMetadataConfigs(JumpStartDataHolderType):
+    """Data class to hold the set of JumpStart Metadata configs."""
+
+    __slots__ = ["configs", "config_rankings", "scope"]
+
+    def __init__(
+        self,
+        configs: Optional[Dict[str, JumpStartMetadataConfig]],
+        config_rankings: Optional[Dict[str, JumpStartConfigRanking]],
+        scope: JumpStartScriptScope = JumpStartScriptScope.INFERENCE,
+    ):
+        """Initializes a JumpStartMetadataConfigs object.
+
+        Args:
+            configs (Dict[str, JumpStartMetadataConfig]):
+                List of configs that the current model has.
+            config_rankings (JumpStartConfigRanking):
+                Config ranking class represents the ranking of the configs in the model.
+            scope (JumpStartScriptScope):
+                The scope of the current config (inference or training)
+        """
+        self.configs = configs
+        self.config_rankings = config_rankings
+        self.scope = scope
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartMetadataConfigs object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    def get_top_config_from_ranking(
+        self,
+        ranking_name: str = JumpStartConfigRankingName.DEFAULT,
+        instance_type: Optional[str] = None,
+    ) -> JumpStartMetadataConfig:
+        """Gets the best the config based on config ranking."""
+        if self.configs and (
+            not self.config_rankings or not self.config_rankings.get(ranking_name)
+        ):
+            raise ValueError("Config exists but missing config ranking.")
+
+        if self.scope == JumpStartScriptScope.INFERENCE:
+            instance_type_attribute = "supported_inference_instance_types"
+        elif self.scope == JumpStartScriptScope.TRAINING:
+            instance_type_attribute = "supported_training_instance_types"
+        else:
+            raise ValueError(f"Unknown script scope {self.scope}")
+
+        rankings = self.config_rankings.get(ranking_name)
+        for config_name in rankings.rankings:
+            resolved_config = self.configs[config_name].resolved_config
+            if instance_type and instance_type not in getattr(
+                resolved_config, instance_type_attribute
+            ):
+                continue
+            return self.configs[config_name]
+
+        return None
+
+
+class JumpStartModelSpecs(JumpStartMetadataBaseFields):
+    """Data class JumpStart model specs."""
+
+    slots = [
+        "inference_configs",
+        "inference_config_components",
+        "inference_config_rankings",
+        "training_configs",
+        "training_config_components",
+        "training_config_rankings",
+    ]
+
+    __slots__ = JumpStartMetadataBaseFields.__slots__ + slots
+
+    def __init__(self, spec: Dict[str, Any]):  # pylint: disable=super-init-not-called
```

Comment:
ack please, call the parent constructor in every code paths, otherwise the code is counter-intuitive.